src/main/java/com/github/dcysteine/neicustomdiagram/generators/gregtech5/recipedebugger/RecipeHandler.java

```diff
@@ -411,4 +419,53 @@ private static boolean unequalCellRecipe(Recipe recipe) {
 
         return countCells(recipe.inputs()) != countCells(recipe.outputs());
     }
+
+    private static boolean SmallVariantRecipe(Recipe recipe) {
+        if (recipe.recipeMap() == RecipeMap.PACKAGER
+                || recipe.recipeMap() == RecipeMap.UNPACKAGER
+                || recipe.recipeMap() == RecipeMap.MACERATOR
+                || recipe.recipeMap() == RecipeMap.LATHE
+                || recipe.recipeMap() == RecipeMap.FLUID_EXTRACTOR
+                || recipe.recipeMap() == RecipeMap.IMPLOSION_COMPRESSOR
+                || recipe.recipeMap() == RecipeMap.ALLOY_SMELTER) {
+            // These recipemaps are meant to have tiny / small dusts or nuggets.
+            return false;
+        }
+        if (recipe.recipeMap() == RecipeMap.ASSEMBLING_MACHINE) {
```

Comment:
I think that this section could be simplified a bit, and the helper method changed a bit.
Something like this (disclaimer: I have not tried compiling this, so it might have mistakes):

```java
private static final ImmutableSet<OrePrefix> SMALL_VARIANT_ORE_PREFIXES =
    ImmutableSet.of(OrePrefixes.dustTiny, OrePrefixes.dustSmall, OrePrefixes.nugget);
private static final ImmutableSet<OrePrefix> CABLE_ORE_PREFIXES =
    ImmutableSet.of(/* ... */);

// Nit: please lowercase the first letter,
// as the convention is that method names should start with a lowercase letter
private static boolean smallVariantRecipe(Recipe recipe) {
  /* Recipe map check */

  Set<OrePrefix> orePrefixes = getOrePrefixes(recipe.outputs().keySet());
  if (recipe.recipeMap() == RecipeMap.ASSEMBLING_MACHINE
      && Sets.intersection(orePrefixes, CABLE_ORE_PREFIXES).size() > 0) {
    // Allow using small dusts for cable insulation.
    return false;
  } else {
    orePrefixes.addAll(getOrePrefixes(recipe.inputs().keySet()));
    return Sets.intersection(orePrefixes, SMALL_VARIANT_ORE_PREFIXES).size() > 0;
  }
}

private static Set<OrePrefix> getOrePrefixes(Set<Component> componentSet) {
  return componentSet.stream()
    .map(GregTechOreDictUtil::getItemData)  // Checks for ComponentType.ITEM for us
    .filter(Optional::isPresent)
    .map(itemData -> itemData.get().mPrefix)
    .collect(Collectors.toCollection(() -> new EnumSet<OrePrefix.class>()));
}
```

Reply:
Oh `Sets.intersection` is actually useful, I'll keep that in mind. And yes I know the convention about naming in Java, it's caused by my careless renaming of method, from `tinySmallDustRecipe` iirc